transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystemRootResourceDefinition.java

```diff
@@ -489,14 +499,19 @@ public DefaultTimeoutHandler(final AttributeDefinition... definitions) {
             super(definitions);
         }
 
-
         @Override
         protected boolean applyUpdateToRuntime(final OperationContext context, final ModelNode operation,
                                                final String attributeName, final ModelNode resolvedValue,
                                                final ModelNode currentValue, final HandbackHolder<Void> handbackHolder)
             throws OperationFailedException {
-            TxControl.setDefaultTimeout(resolvedValue.asInt());
-            ContextTransactionManager.setGlobalDefaultTransactionTimeout(resolvedValue.asInt());
+            int timeout = resolvedValue.asInt();
+
+            TxControl.setDefaultTimeout(timeout);
+            if (timeout == 0) {
+                timeout = TransactionSubsystemRootResourceDefinition.maximum_timeout;
```

Comment:
Rather than using this static field, you should read and resolve from the model. Something like
````
MAXIMUM_TIMEOUT.resolveModelAttribute(context, context.readResource(operation.currentAddress())
````

Reply:
I tried
```
ModelNode model = context.readResource(context.getCurrentAddress()).getModel();
timeout = MAXIMUM_TIMEOUT.resolveModelAttribute(context, model).asInt();
```
but it fails with
```
21:06:59,820 ERROR [org.jboss.as.controller.management-operation] (management-handler-thread - 1) WFLYCTL0013: Operation ("write-attribute") failed - address: ([("subsystem" => "transactions")]) - failure description: "WFLYCTL0071: Duplicate path element 'subsystem' found"

```

Reply:
It has been a while since I wrote code in this area :) I remember now that `context.readResource()` is relative to the operation address, so try `context.readResource(PathAddress.EMPTY_ADDRESS)`